src/main/java/core/basesyntax/operationhandler/SupplyOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.operationhandler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransfer;
+
+public class SupplyOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransfer transfer) {
+        Fruit fruit = transfer.getFruit();
+        Integer currentValue = Storage.storageMap.get(fruit);
+        if (currentValue == null) {
+            throw new RuntimeException("current value can't be null");

```

Comment:
Ну почему же throw new RuntimeException???
Привезли фрукты, которых еще не было на складе, вот и все.
```suggestion
            currentValue = 0;
```

**А еще лучше не воспользоваться советом выше, а узнать что-то новое про интерфейс Map и заюзать метод merge()**